qualibration_graphs/superconducting/calibrations/CZ_calibration_fixed_couplers/20b_cz_conditional_phase_error_amp.py

```diff
@@ -0,0 +1,302 @@
+1. The phase difference is closest to π (0.5 in normalized units)
+2. The leakage to the |f> state is minimized
+
+Prerequisites:
+- Calibrated single-qubit gates for both qubits in the pair
+- Calibrated readout for both qubits
+- Initial estimate of the CZ gate amplitude
+
+State update:
+- The optimal CZ gate amplitude: qubit_pair.gates["Cz"].flux_pulse_control.amplitude
+"""
+
+# Be sure to include [Parameters, Quam] so the node has proper type hinting
+node = QualibrationNode[Parameters, Quam](
+    name="20b_cz_conditional_phase_error_amp",  # Name should be unique
+    description=description,  # Describe what the node is doing, which is also reflected in the QUAlibrate GUI
+    parameters=Parameters(),  # Node parameters defined under calibration_utils/cz_conditional_phase/parameters.py
+)
+
+
+# Any parameters that should change for debugging purposes only should go in here
+# These parameters are ignored when run through the GUI or as part of a graph
+@node.run_action(skip_if=node.modes.external)
+def custom_param(node: QualibrationNode[Parameters, Quam]):
+    # node.parameters.qubit_pairs = ["q1-q2"]
+    pass
+
+
+# Instantiate the QUAM class from the state file
+node.machine = Quam.load()
+
+
+# %% {Create_QUA_program}
+@node.run_action(skip_if=node.parameters.load_data_id is not None)
+def create_qua_program(node: QualibrationNode[Parameters, Quam]):
+    """Create the sweep axes and generate the QUA program from the pulse sequence and the node parameters."""
+
+    # Class containing tools to help handle units and conversions.
+    u = unit(coerce_to_integer=True)
+    # Get the active qubit pairs from the node and organize them by batches
+    node.namespace["qubit_pairs"] = qubit_pairs = get_qubit_pairs(node)
+    num_qubit_pairs = len(qubit_pairs)
+
+    # Extract the sweep parameters and axes from the node parameters
+    n_avg = node.parameters.num_averages
+    amplitudes = np.arange(1 - node.parameters.amp_range, 1 + node.parameters.amp_range, node.parameters.amp_step)
+    frames = np.arange(0, 1, 1 / node.parameters.num_frames)
+
+    operation_name = node.parameters.operation
+    num_operations = node.parameters.number_of_operations
+    # Register the sweep axes to be added to the dataset when fetching data
+    node.namespace["sweep_axes"] = {
+        "qubit_pair": xr.DataArray(qubit_pairs.get_names()),
+        "number_of_operations": xr.DataArray(
+            np.arange(1, num_operations + 1),
+            attrs={"long_name": "number of operations"},
+        ),
+        "amp": xr.DataArray(amplitudes, attrs={"long_name": "amplitude scale", "units": "a.u."}),
+        "frame": xr.DataArray(frames, attrs={"long_name": "frame rotation", "units": "2π"}),
+        "control_axis": xr.DataArray([0, 1], attrs={"long_name": "control qubit state"}),
+    }
+
+    # The QUA program stored in the node namespace to be transfer to the simulation and execution run_actions
+    with program() as node.namespace["qua_program"]:
```

Comment:
I think that it would be nice to have the same comments and parameter names as for the single CZ (node 20a) for consistency. For instance, here the operation is called `operation_name` instead of `operation` in node 20a. I think that everything should be identical except the loop over applying CZ and the frame rotation trick.